lib/core/src/Cardano/Wallet.hs

```diff
@@ -1457,11 +1476,80 @@ balanceTransaction
     (pp, nodePParams)
     ti
     (internalUtxoAvailable, wallet, pendingTxs)
-    (PartialTx partialTx externalInputs redeemers)
+    (PartialTx partialTx@(cardanoTx -> Cardano.InAnyCardanoEra _ (Cardano.Tx (Cardano.TxBody bod) _)) externalInputs redeemers)
     = do
     let (outputs, txWithdrawal, txMetadata, txAssetsToMint, txAssetsToBurn)
             = extractFromTx partialTx
 
+    -- Coin selection does not support pre-defining collateral. In Sep 2021
+    -- consensus was that we /could/ allow for it with just a day's work or so,
+    -- but that the need for it was unclear enough that it was not in any way
+    -- a priority.
+    case Cardano.txInsCollateral bod of
+        Cardano.TxInsCollateralNone -> return ()
+        Cardano.TxInsCollateral _ [] -> return ()
+        Cardano.TxInsCollateral _ _ ->
+            throwE $ ErrBalanceTxNotYetSupported ExistingCollateral
```

Comment:
This is not expected to change, so the wrapping `ErrBalanceTxNotYetSupported ` might be inappropriate. It should deserve a top-level constructor.